src/utils/process.rs

```diff
@@ -630,22 +705,78 @@ pub mod tests {
     #[test]
     #[should_panic]
     fn test_process_testing_assert() {
-        {
-            let _args = cfg::WithArgs::new(&"git blame do.not.panic");
-            assert_eq!(
-                calling_process_cmdline(ProcInfo::new(), blame::guess_git_blame_filename_extension),
-                Some("panic".into())
-            );
+        let _args = FakeParentArgs::once(&"git blame do.not.panic");
+        assert_eq!(
+            calling_process_cmdline(ProcInfo::new(), guess_git_blame_filename_extension),
+            Some("panic".into())
+        );
 
-            calling_process_cmdline(ProcInfo::new(), blame::guess_git_blame_filename_extension);
-        }
+        calling_process_cmdline(ProcInfo::new(), guess_git_blame_filename_extension);
+    }
```

Comment:
I think that this module is getting to the point where it would be easier to understand if we separated the abstract machinery (e.g. `ProcessInterface`, `calling_process_cmdline` , `FakeParentArgs`) from the concrete clients (`determine_calling_process`, `git_blame_filename_extension`). E.g. I feel that this test is starting to mix up testing `git blame` and `FakeParentArgs` logic. Happy for that to be done in subsequent PRs though.